### DISCUSSION TITLE
How to get data from other tables

### DISCUSSION THREAD

**Discussion by jairiza** [AntlerVC/firetable#1325] [Support / Q&A] [answered]
I am trying to run a formula on a column in one table using fields from another column in a different table. How can I get access to the data in that column in the other table?

**Comment by BeeBombshell** [comment, marked as answer]
Hey @jairiza, you could use the Derivative field type to achieve this. Derivative fields have access to parameters that can access your data on Firestore (check out our docs to know more: https://docs.rowy.io/field-types/derivative#api).

The following code snippet demonstrates fetching data from a "users" collection in your Firestore database.

```js
const derivative = async ({ rrow, ref, db, storage, auth }) => {
  // Get user doc
  const userDoc = await db.collection("users").doc(row.uid).get(); // matches the uid column in this table to the Document ID in users collection
  return userDoc.get("email");
};

export default derivative;
```

> To know more about setting custom Document IDs, check: https://docs.rowy.io/how-to/add-row#with-custom-id and https://docs.rowy.io/import-export-data/import-csv#autogenerated-id-vs-custom-id
